src/sdk/namespace_fs.js

```diff
@@ -2618,12 +2618,31 @@ class NamespaceFS {
         }
     }
 
+    /**
+     * _delete_path_dirs deletes all the paths in the hierarchy that are empty after a successful delete
+     * if the original file_path to be deleted is a regular object which means file_path is not a directory and it's not a directory object path - 
+     * before deletion of the parent directory  - 
+     * if the parent directory is a directory object (has CONTENT_DIR xattr) - stop the deletion loop
+     * else - delete the directory - if dir is not empty it will stop at the first non empty dir
+     * NOTE - the directory object check is needed because when object size is zero we won't create a .folder file and the dir will be empty
+     * therefore the deletion will succeed although we shouldn't delete the directory object
+     * @param {String} file_path 
+     * @param {nb.NativeFSContext} fs_context 
+     */
     async _delete_path_dirs(file_path, fs_context) {
         try {
-            let dir = path.dirname(file_path);
-            while (dir !== this.bucket_path) {
-                await nb_native().fs.rmdir(fs_context, dir);
-                dir = path.dirname(dir);
+            let dir_path = path.dirname(file_path);
+            while (dir_path !== this.bucket_path) {
+                const file_is_dir = file_path === path.join(dir_path, '/');
+                const file_is_dir_content_hidden_file = file_path === path.join(dir_path, config.NSFS_FOLDER_OBJECT_NAME);
+                const file_is_regular_obj = !file_is_dir && !file_is_dir_content_hidden_file;
+                if (file_is_regular_obj) {
+                    const dir_stat = await nb_native().fs.stat(fs_context, dir_path);
+                    const file_is_disabled_dir_content = dir_stat.xattr && dir_stat.xattr[XATTR_DIR_CONTENT] !== undefined;
+                    if (file_is_disabled_dir_content) break;
+                }
+                await nb_native().fs.rmdir(fs_context, dir_path);
+                dir_path = path.dirname(dir_path);
             }
```

Comment:
I think what confuses me about this is that this condition can be true only for the first iteration. maybe you can assign it before something like:
```suggestion
            let dir_path = path.dirname(file_path);
            const file_is_dir = file_path.endsWith('/') || file_path.endsWith(config.NSFS_FOLDER_OBJECT_NAME);
            const first = true;
            while (dir_path !== this.bucket_path) {
                if (first && !file_is_dir) {
                const dir_stat = await nb_native().fs.stat(fs_context, dir_path);
                    const file_is_disabled_dir_content = dir_stat.xattr && dir_stat.xattr[XATTR_DIR_CONTENT] !== undefined;
                    if (file_is_disabled_dir_content) break;
                }
                await nb_native().fs.rmdir(fs_context, dir_path);
                dir_path = path.dirname(dir_path);
                first = false;
```
hopefully I understand the intention correctly. its still not ideal with the first argument always assigned to be false. but maybe easier to understand? WDYT?

Reply:
another thing to consider is that I think dirname of file that ends with "/" is to directory above it. so maybe file_path.endsWith('/') will cause problems. if this is correct, it worked in the first time because  `file_is_dir = file_path === path.join(dir_path, '/');` can never be true. again sorry if I am completely wrong

Reply:
it's the opposite, only on the first iteration we might want to skip the check and delete the folder directly (if the file is a directory object)
on all the other iterations we must check if the folder is a directory object

I'll try a different approach so you can understand it better